LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2Effect_CapStat.uc

```diff
@@ -0,0 +1,163 @@
+//--------------------------------------------------------------------------------------- 
+//  FILE:    X2Effect_CapStat.uc
+//  AUTHOR:	 Musashi, modified by Grobobobo
+//  PURPOSE: Creates an effect that can cap a stat to a certain point
+//--------------------------------------------------------------------------------------- 
+
+class X2Effect_CapStat extends X2Effect_PersistentStatChange;
+
+var array<X2Condition> Conditions;
+
+struct StatCap
+{
+	var ECharStatType StatType;
+	var float StatCapValue;
+	var bool Minimum; // If true, the cap means that the stat cannot go lower than the amount, if false, the higher
+};
+
+var array<StatCap> StatCaps; 
+
+function AddStatCap(ECharStatType StatType, float StatCapValue, bool Minimum)
+{
+	local StatCap Cap;
+	
+	Cap.StatType = StatType;
+	Cap.StatCapValue = StatCapValue;
+	Cap.Minimum = Minimum;
+	StatCaps.AddItem(Cap);
+}
+
+function RegisterForEvents(XComGameState_Effect EffectGameState)
+{
+	local XComGameState_Unit UnitState;
+	local X2EventManager EventMgr;
+	local Object ListenerObj;
+
+	EventMgr = `XEVENTMGR;
+
+	UnitState = XComGameState_Unit(`XCOMHISTORY.GetGameStateForObjectID(EffectGameState.ApplyEffectParameters.TargetStateObjectRef.ObjectID));
+
+	ListenerObj = EffectGameState;
+
+	// Register to tick after EVERY action.
+	EventMgr.RegisterForEvent(ListenerObj, 'OnUnitBeginPlay', EventHandler, ELD_OnStateSubmitted, 25, UnitState,, EffectGameState);	
+	//EventMgr.RegisterForEvent(ListenerObj, 'AbilityActivated', EventHandler, ELD_OnStateSubmitted, 25,,, EffectGameState);
+	EventMgr.RegisterForEvent(ListenerObj, 'UnitTakeEffectDamage', EventHandler, ELD_OnStateSubmitted, 25,,, EffectGameState);	
```

Comment:
Why did you change this? `UnitTakeEffectDamage` makes this approach even less reliable, since it will never proc for abilities that don't damage but do apply stat buffs/debuffs. This was the whole reason for me suggesting Musashi's implementation.